packages/paste-core/components/modal/src/Modal.tsx

```diff
@@ -123,22 +129,30 @@ const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
         {transitions(
           (styles, item) =>
             item && (
-              <ModalDialogOverlay
+              <Box
+                // @ts-expect-error Render overlay as box for customization
```

Comment:
I actually don't _mind_ this approach, but if there is some push back you can achieve it with styled components. We did it in Textarea https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/textarea/src/TextArea.tsx#L71

The custom css function is exported from Box, and you can pass it to the styled component manually to merge in the custom styles from `element`. You just need to manually apply the `data-paste-element` attributes on the node you want to target. 

I don't have a preference on either way. This is just knowledge share